### PR TITLE
Adapt 241

### PR DESCRIPTION
**Pull request by w-aldrich**
Finally ready for a PR! 🥳 
This adds the addition of tags, with an authenticated tags endpoint (to create) non authentication for getting all tags
Tags are stored as a list of strings in the metadata topic, but tags are stored as key(string) value(string) in the tags topic. Where the key is the tag, and the value is the description of the tag.

### CODE REVIEW

File: ingestors/kafka/src/main/scala/hydra/kafka/endpoints/TagsEndpoint.scala [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+  override val route: Route = {
+    extractMethod { method =>
+      handleExceptions(exceptionHandler(Instant.now, method.value)) {
+        extractExecutionContext { implicit ec =>
+          pathPrefix("v2" / "tags") {
+            get {
+              onComplete(Futurable[F].unsafeToFuture(tagsAlgebra.getAllTags)) {
+                case Failure(exception) => complete(StatusCodes.InternalServerError, exception)
```
Comment (w-aldrich) [inline ingestors/kafka/src/main/scala/hydra/kafka/endpoints/TagsEndpoint.scala:34]: add http metrics

File: ingestors/kafka/src/test/scala/hydra/kafka/algebras/TagsAlgebraSpec.scala [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        val tag = HydraTag("Source: blah", "This comes from blah")
+        tagsAlgebra.createOrUpdateTag(tag)
+          .unsafeRunSync shouldBe Right(PublishResponse(0,1))
+      }
+      "Create a different Tag" in {
+        val tag = HydraTag("Source: Bret", "This comes from Bret being on a call while I do this")
+        tagsAlgebra.createOrUpdateTag(tag)
+          .unsafeRunSync shouldBe Right(PublishResponse(0,2))
+      }
+      "Get Tags" in {
```
Comment (w-aldrich) [inline ingestors/kafka/src/test/scala/hydra/kafka/algebras/TagsAlgebraSpec.scala:42]: Possibly just move to e2e

Comment (w-aldrich) [reply ingestors/kafka/src/test/scala/hydra/kafka/algebras/TagsAlgebraSpec.scala:42]: Moved entire class to e2e testing

File: ingest/src/main/scala/hydra.ingest/app/AppConfig.scala [hunk 3]
```diff
@@ -162,14 +162,39 @@ object AppConfig {
...
-      topicDeletionConfig: TopicDeletionConfig,
-      dvsConsumersTopicConfig: DVSConsumersTopicConfig,
-      consumerOffsetsOffsetsTopicConfig: ConsumerOffsetsOffsetsTopicConfig,
-      consumerGroupsAlgebraConfig: ConsumerGroupsAlgebraConfig
+                              createTopicConfig: CreateTopicConfig,
+                              metadataTopicsConfig: MetadataTopicsConfig,
+                              ingestConfig: IngestConfig,
+                              topicDeletionConfig: TopicDeletionConfig,
+                              tagsPasswordConfig: TagsEndpointPasswordConfig,
+                              tagsTopicConfig: TagsTopicConfig,
```
Comment (bretrbs) [inline ingest/src/main/scala/hydra.ingest/app/AppConfig.scala:193]: Do we need these 3 configs to be separate?

File: ingest/src/main/scala/hydra.ingest/app/AppConfig.scala [hunk 4]
```diff
@@ -162,14 +162,39 @@ object AppConfig {
...
+  private val tagsTopicConfig: ConfigValue[TagsTopicConfig] =
+    (
+      env("TAGS_TOPIC").as[String].default("_hydra.tags-topic")
+    ).map(TagsTopicConfig)
+
+  final case class TagsConsumerGroupConfig(tagsConsumerGroup: String)
+
+  private val tagsConsumerGroupConfig: ConfigValue[TagsConsumerGroupConfig] =
+    (
+      env("TAGS_CONSUMER_GROUP").as[String].default("_hydra.tags-consumer-group")
```
Comment (w-aldrich) [inline ingest/src/main/scala/hydra.ingest/app/AppConfig.scala:183]: Should be unique consumer group per node. Send in as a config.

File: ingestors/kafka/src/main/scala/hydra/kafka/endpoints/TagsEndpoint.scala [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        }
+      }
+    }
+  }
+
+  private def exceptionHandler(startTime: Instant, method: String) = ExceptionHandler {
+    case e =>
+      extractExecutionContext{ implicit ec =>
+        addHttpMetric("", StatusCodes.InternalServerError,"/v2/tags", startTime, method, error = Some(e.getMessage))
+        complete(500, e.getMessage)
```
Comment (w-aldrich) [inline ingestors/kafka/src/main/scala/hydra/kafka/endpoints/TagsEndpoint.scala:67]: StatusCodes

File: ingestors/kafka/src/test/scala/hydra/kafka/model/TopicMetadataSpec.scala [hunk 6]
```diff
@@ -115,16 +119,19 @@ final class TopicMetadataSpec extends AnyFlatSpecLike with Matchers {
   }
 
   "TopicMetadata" should "have compatible schema evolutions" in {
+    // ADD new schema
```
Comment (w-aldrich) [inline ingestors/kafka/src/test/scala/hydra/kafka/model/TopicMetadataSpec.scala:122]: Remove

**Pull request merged**